llvm/cmake/modules/AddLLVM.cmake

```
@@ -2211,6 +2211,17 @@ function(add_lit_testsuites project directory)
       if (NOT (is_inputs EQUAL -1 AND is_output EQUAL -1))
         continue()
       endif()
+      set(_should_skip FALSE)
+      foreach(skip ${ARG_SKIP})
+        string(FIND ${lit_suite} ${skip} is_skip)
+        if (NOT is_skip EQUAL -1)
```

```suggestion
  foreach(skip IN LISTS ARG_SKIP)
    if(lit_suite MATCHES "${skip}")
      set(_should_skip TRUE)
```

Would that work?

(You're using FIND which I'm not sure if an exact match)

---

i moved the check down to after `name_slash` (and made the check on `name_slash MATCHES skip`) so now i think we have the effect you wanted (ability to precisely match using regex).